### PR TITLE
bpo-46838: Syntax error improvements for function definitions

### PR DESCRIPTION
**Pull request by pablogsal**
First round of improvements for function definitions.

https://bugs.python.org/issue46838

### CODE REVIEW

File: Misc/NEWS.d/next/Core and Builtins/2022-02-25-22-42-30.bpo-46838.RB6kEy.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Improve yntax errors for incorrect function definitions. Patch by Pablo
```
Comment (AlexWaygood) [inline Misc/NEWS.d/next/Core and Builtins/2022-02-25-22-42-30.bpo-46838.RB6kEy.rst]:
```suggestion
Improve syntax errors for incorrect function definitions. Patch by Pablo
```

**Comment by pablogsal** [comment]
Thanks @AlexWaygood !!

**Comment by Andykl** [comment]
It's great that all the points for function defenition have worked out.
1. Do the new rules apply to lambda function parameters?
2. What will be for `def foo(arg1, */, arg2): pass` (*/, not /*)? It looks like this case is not covered.

**Comment by pablogsal** [comment]
> * Do the new rules apply to lambda function parameters?

No

> What will be for def foo(arg1, */, arg2): pass (*/, not /*)? It looks like this case is not covered.

I left it uncovered because that's going to be quite rare and I think is not worth the extra rule

**Pull request merged**